pkg/sql/logictest/testdata/logic_test/privilege_builtins

```diff
@@ -1056,3 +1056,53 @@ query B
 SELECT has_column_privilege('hcp_test'::REGCLASS, 4, 'SELECT')
 ----
 true
+
+# Regression test for #58254.
+
+user root
+
+statement ok
+DROP DATABASE IF EXISTS my_db;
+CREATE DATABASE my_db;
+
+statement ok
+CREATE ROLE my_role;
+
+statement ok
+GRANT ALL ON DATABASE my_db TO my_role;
```

Comment:
just to exercise more of the logic, maybe this test could just `GRANT SELECT ON DATABASE my_db TO my_role`? and further down, it could `GRANT ALL ON DATABASE my_db TO testuser2`.

then it could check that testuser has SELECT privilege on my_db but does not have CREATE privilege, while testuser2 has both.

(and similar for the user-defined schema that you make as per my earlier comment)

the idea here is to make sure that privilege inheritance is additive -- testuser2 should also be able to get its own privileges separately from whatever it inherits.

Reply:
I've done this for the user-defined schema, but had trouble confirming for the database privileges because the only options for [`has_database_privilege` builtin are CREATE, CONNECT, TEMPORARY, and TEMP](https://github.com/cockroachdb/cockroach/blob/c18a8d3fdc9828ca154ab95634b2c7f0c2fe0477/pkg/sql/sem/builtins/pg_builtins.go#L1320). Of these, only CREATE can be granted. If I grant CREATE, then testing the remaining 3 options results in the following:
- CONNECT always returns true as the comment states all users have CONNECT privilege for all databases.
- TEMPORARY and TEMP follow the same privilege check under the hood as CREATE.

So it doesn't seem possible to test one privilege but not another for this built-in on a database.

Reply:
interesting, i didn't know that's how has_database_privilege worked! no need to test that here then